sys/mips/pmap.c

```diff
@@ -547,3 +549,34 @@ void pmap_delete(pmap_t *pmap) {
   free_asid(pmap->asid);
   pool_free(P_PMAP, pmap);
 }
+
+#define L1_SPACE_SIZE (PAGESIZE * PAGESIZE / sizeof(pte_t))
+
+/*
+ * Increase usable kernel virtual address space to at least maxkvaddr.
+ * Allocate page table (level 1) if needed.
+ */
+vaddr_t pmap_growkernel(vaddr_t maxkvaddr) {
```

Comment:
This function modifies the global variable `pmap_maxkvaddr` without any synchronization. We're holding `maxkvaddr_lock` while calling it, but it's not required. Could you add an assertion here that checks that we're holding `maxkvaddr_lock`?

Reply:
My bad, we're holding `pmap_kernel()->mtx` when modifying `pmap_maxkvaddr`, but some accesses are unlocked, so we need to make them atomic to avoid tripping up KCSAN.

Reply:
done